test/jdk/java/net/httpclient/ResponsePublisher.java

```diff
@@ -451,6 +492,23 @@ public void teardown() throws Exception {
         }
     }
 
+    // Wait for the client to be garbage collected.
+    // we use the ReferenceTracker API rather than HttpClient::close here,
+    // because these tests inject faults by throwing inside callbacks, which
+    // is more likely to get HttpClient::close wedged until jtreg times out.
+    // By using the ReferenceTracker, we will get some diagnosis about what
+    // is keeping the client alive if it doesn't get GC'ed within the
+    // expected time frame.
+    public void clientCleanup(OperationTrackers.Tracker tracker){
+        System.gc();
+        System.out.println(now() + "waiting for client to shutdown: " + tracker.getName());
+        System.err.println(now() + "waiting for client to shutdown: " + tracker.getName());
+        var error = TRACKER.check(tracker, 10000);
```

Comment:
Hello Darragh, 10 seconds looks a bit too large, even if it's the max time to wait. Before this change the graceDelayms was 500 milli seconds. So this appears to be a big jump. Perhaps we should start with something like 2 or 5 seconds (an arbitrary number, I admit)?

Reply:
It shouldn't take 10 seconds for the client to shutdown, so it shouldn't matter that we use a large timeout. I we used HttpClient::close() we would wait forever (that is - we would wait as long as it takes).
I believe 500ms to wait for the GC to kick in was a bit optimistic for cases where the machine is overloaded by other tasks. Waiting for a client to be gc'ed or closed until we create the next one should help, as it will ensure there are no lingering resources waiting to be released. 
The advantage of using close() is that it is much cleaner and simpler. The disadvantage is that if it fails in timeout there, you have no clue about what prevented close() to finish. In opposition the tracker will provide diagnosis if the client fails to close in the imparted delay.

Reply:
>The advantage of using close() is that it is much cleaner and simpler. The disadvantage is that if it fails in timeout there, you have no clue about what prevented close() to finish. In opposition the tracker will provide diagnosis if the client fails to close in the imparted delay.

That sounds reasonable to me. The changes look fine in this PR.